src/session.ts

```diff
@@ -155,22 +155,30 @@ export class Session extends Client implements ErdstallSession {
 		return (this.erdstallConn as LedgerWriter).withdraw(exitProof);
 	}
 
-	async leave(): Promise<TransactionGenerator> {
+	async leave(
+		notify: (message: string) => void,
```

Comment:
```suggestion
		notify?: (name: string, stage: number, maxStages: number) => void,
```
Change the meaning of notify to inform about the beginning of stages, not when they're done.